### PR TITLE
Update recommendations.yaml

### PR DESCRIPTION
**Pull request by bkaage**
Updated by Bkaage on 5/23/24

# Overview/Summary

Updated content in Key Vault recommendations. Added 1 new recommendation, Removed Soft Delete and marked all as PG Verified - True

## This PR fixes/adds/changes/removes

1. Updated content in Key Vault recommendations. Added 1 new recommendation, Removed Soft Delete and marked all as PG Verified - True

### Breaking Changes

None

## As part of this pull request I have

- [ ] Read the [Contribution Guide](https://azure.github.io/Azure-Proactive-Resiliency-Library-v2/contributing) and ensured this PR is compliant with the guide
- [ ] Checked for duplicate [Pull Requests](https://github.com/Azure/Azure-Proactive-Resiliency-Library-v2/pulls)
- [ ] Associated it with relevant [GitHub Issues](https://github.com/Azure/Azure-Proactive-Resiliency-Library-v2/issues) or ADO Work Items (Internal Only)
- [ ] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/Azure-Proactive-Resiliency-Library-v2/tree/main)
- [ ] Ensured PR tests are passing
- [ ] Performed testing and provided evidence (e.g. screenshot of output) for any changes associated to ARG queries
- [ ] Updated relevant and associated documentation (e.g. Contribution Guide, Docs etc.)

### CODE REVIEW

**Comment by Dost2010** [comment]
@bkaage  | limit potentialbenefit field is 60 characters. For more information please refer to contribution guide --> https://azure.github.io/Azure-Proactive-Resiliency-Library-v2/contributing/create-content/create-recommendations/

File: azure-resources/KeyVault/vaults/recommendations.yaml [hunk 1]
```diff
@@ -1,21 +1,21 @@
-- description: Key vaults should have soft delete enabled
-  aprlGuid: 1cca00d2-d9ab-8e42-a788-5d40f49405cb
+- description: Azure Key Vault RBAC for Data Plane
```
Comment (oZakari) [inline azure-resources/KeyVault/vaults/recommendations.yaml:1]: Could you please remove the soft-delete kql file and then add a new kql file with the new recommendation guide.

**Review by oZakari — CHANGES_REQUESTED** [review]

File: azure-resources/KeyVault/vaults/recommendations.yaml [hunk 2]
```diff
@@ -1,21 +1,21 @@
...
-  recommendationControl: Disaster Recovery
+  recommendationControl: Security
   recommendationImpact: High
   recommendationResourceType: Microsoft.KeyVault/vaults
   recommendationMetadataState: Active
   longDescription: |
-    Key Vault's soft-delete feature enables recovery of deleted vaults and objects like keys, secrets, and certificates. When enabled, marked resources are retained for 90 days, allowing for their recovery, essentially undoing deletion.
-  potentialBenefits: Enables recovery of deleted items
-  pgVerified: false
+    Enhances security by integrating with Azure RBAC, replacing the legacy Access Policy model.
```
Comment (oZakari) [inline azure-resources/KeyVault/vaults/recommendations.yaml:9]:
```suggestion
   Boosts security by integrating with Azure RBAC, replacing Access Policy
```

Needs to be less than 60 characters

**Comment by bkaage** [comment]
Please review

File: azure-resources/KeyVault/vaults/recommendations.yaml (hunk 1 above)
Comment (oZakari) [reply azure-resources/KeyVault/vaults/recommendations.yaml:1]: It doesn't look like the KQL file was removed for soft delete. There also wasnt a new query file added for the new data plane recommendation.

Also, could you please reach out to me in Teams (ztrocinski) so I can confirm your Microsoft alias.

**Pull request closed without merging**